port/unix/omrsock.c

```diff
@@ -82,25 +82,50 @@ omrsock_getaddrinfo(struct OMRPortLibrary *portLibrary, char *node, char *servic
 int32_t
 omrsock_getaddrinfo_length(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t handle, uint32_t *length)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	*length = handle->length;
+	return 0;
 }
 
 int32_t
 omrsock_getaddrinfo_family(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t handle, int32_t *family, int32_t index)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	omr_os_addrinfo *info = handle->addrInfo;
+	int32_t i = 0;
+
+	for(i = 0; i < index; i++) {
+		info = info->ai_next;
```

Comment:
is `info->ai_next` or `info` guaranteed to be non-null? dereferencing a null pointer will cause a seg-fault. there should be  a null check:
```suggestion
		if (info != null) {
		 	 info = info->ai_next;
		}
```
you should also consider error handling in case `info` is null.

Reply:
@babsingh Sounds good. I'm going to check if info is null, which would be an error (I can't really check it in the test cases I think because the users can't directly access it). I'm going to also check if handle is NULL, and if index is greater than length, which both would be argument errors (These two will be in the test cases).

For elements of info->ai_next and ai_family and so on, they could be NULL or 0 but they will exist for sure. With the length parameter, as long as index is smaller than length, we shouldn't run into the problem of accessing a NULL pointer.